app/serializers/datacite_doi_serializer.rb

```diff
@@ -119,24 +119,19 @@ class DataciteDoiSerializer
   attribute :doi, &:uid
 
   attribute :publisher do |object, params|
-    # new - format obj only if ?publisher=true, otherwise serialize the old format (a string)
-    if object.publisher.nil?
-      nil
+    # publisher accessor will now always return a publisher object, and indexed documents will store a "publisher_obj" attribute with the publisher object
+    # new obj format only if ?publisher=true, otherwise serialize the old format (a string)
+    publisher = object.try(:publisher_obj) || object.try(:publisher)
+
+    nil if publisher.nil?
```

Comment:
we should return `nil` here, if do not say `return nil` code execution will be continued in the next line, 
for example, here I have `test` method which has `nil` without return, if you call this method, you will get "hi".

```
irb(main):008:0> def test
irb(main):009:1> nil
irb(main):010:1> return "hi"
irb(main):011:1> end
=> :test
irb(main):012:0> test
=> "hi"
irb(main):013:0> 
```

Reply:
Thank you for catching this, Ashwini! I was getting `LocalJumpError - unexpected return` with returns—I think because you have to return within a method rather than a loop—so I made some adjustments and fixed the tests that weren't catching this. Let me know what you think.

Reply:
You're welcome! :)